lib/src/components/NxDropdown/stateful/__tests__/NxStatefulDropdown.test.tsx

```diff
@@ -5,192 +5,311 @@
-      container = null;
-    }
-  });
+    label: 'dropdown-toggle',
+    children: <a>Hello</a>
+  };
 
-  it('renders an NxDropdown component', function() {
-    const component = getShallowComponent();
-    expect(component).toMatchSelector(NxDropdown);
-    expect(component).toHaveProp('label', 'dropdown');
-  });
+  const renderEl = rtlRenderElement<Props>(NxStatefulDropdown, minimalProps);
+  const quickRender = rtlRender<Props>(NxStatefulDropdown, minimalProps);
 
-  it('renders an NxDropdown component passing on the given props', function() {
-    const component = getShallowComponent({
+  it('renders an NxStatefulDropdown toggle element correctly with the given props', function() {
+    const { container } = quickRender({
       variant: 'tertiary',
       className: 'extra-class',
-      disabled: true,
-      toggleTooltip: 'a tooltip'
+      disabled: true
     });
-    const statelessDropdown = component.find(NxDropdown);
 
-    expect(component).toMatchSelector(NxDropdown);
-    expect(statelessDropdown).toHaveProp('label', 'dropdown');
-    expect(statelessDropdown).toHaveProp('variant', 'tertiary');
-    expect(statelessDropdown).toHaveProp('disabled', true);
-    expect(statelessDropdown).toHaveProp('toggleTooltip', 'a tooltip');
-    expect(statelessDropdown).toHaveProp('children', (<a>Hello</a>));
+    const dropdown = container.querySelector('.nx-dropdown');
+    const toggleButton = screen.getByRole('button', { name: 'dropdown-toggle'});
+
+    expect(dropdown).toHaveClass('extra-class');
+    expect(toggleButton).toHaveClass('nx-btn--tertiary');
+    expect(toggleButton).toHaveClass('disabled');
+    expect(toggleButton).toHaveAttribute('aria-disabled');
   });
 
-  it('opens the dropdown when the toggle is clicked', function() {
-    const mounted = getMountedComponent(undefined, { attachTo: container });
+  it('toggles the menu open and close when toggle is clicked', async function() {
+    const user = userEvent.setup();
 
-    expect(mounted.find('.nx-dropdown-menu')).not.toExist();
+    const { container } = quickRender();
+    const toggleButton = screen.getByRole('button', { name: 'dropdown-toggle'});
 
-    mounted.find(NxButton).simulate('click');
-    expect(mounted.find('.nx-dropdown-menu')).toExist();
-    mounted.unmount();
+    let menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
+
+    await user.click(toggleButton);
+
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await user.click(toggleButton);
+
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
   });
 
-  it('closes the dropdown if the menu is open and the toggle is clicked', function() {
-    const mounted = getMountedComponent(undefined, { attachTo: container });
+  it('closes the dropdown when the Escape key is pressed on this component', async function() {
```

Comment:
Should probably have a test that pressing ESC on an already-closed dropdown doesn't open it

Reply:
> Should probably have a test that pressing ESC on an already-closed dropdown doesn't open it

I added that test here: a88942639a230546e9e35bdb1d808c316a173f01

I've revised it a little here: ca629a0bcdd4105fd7e590a19f980ad231ebe4f8
Let me know what you think.

Reply:
> I added that test here: https://github.com/sonatype/sonatype-react-shared-components/commit/a88942639a230546e9e35bdb1d808c316a173f01
>
> I've revised it a little here: https://github.com/sonatype/sonatype-react-shared-components/commit/ca629a0bcdd4105fd7e590a19f980ad231ebe4f8
Let me know what you think.

I think your first impl was missing an `await`.  But why'd you switch to `fireEvent` in the second attempt?

Reply:
> > I added that test here: [a889426](https://github.com/sonatype/sonatype-react-shared-components/commit/a88942639a230546e9e35bdb1d808c316a173f01)
> > I've revised it a little here: [ca629a0](https://github.com/sonatype/sonatype-react-shared-components/commit/ca629a0bcdd4105fd7e590a19f980ad231ebe4f8)
> > Let me know what you think.
> 
> I think your first impl was missing an `await`. But why'd you switch to `fireEvent` in the second attempt?

Actually, I'm not sure why. I thought I had a reason to switch to fireEvent 😵 fixed it here:

872e45cc68a5ef5d3e6e4ea8dc7d96a2ddc4ad67